Google.Api.Gax.Grpc.IntegrationTests/ClientBuilderBaseTest.cs

```diff
@@ -340,7 +391,29 @@ private protected override ChannelBase CreateChannel(string endpoint, ChannelCre
                 return ChannelCreated;
             }
 
+            protected override GoogleCredential SetUseJwtWithScopesFlag(GoogleCredential credential)
+            {
+                GoogleCredentialsUsedToCreateChannel = base.SetUseJwtWithScopesFlag(credential);
+
+                return GoogleCredentialsUsedToCreateChannel;
+            }
+
             public override string ToString() => _name;
         }
+
+        public class SampleClientBuilderWithJwt : SampleClientBuilder
```

Comment:
Instad of adding all these different classes, add a constructor to the existing one that takes 2 bools `public SampleClientBuilder(bool clientUsesJwt, bool poolUsesJwt)`, and then you can test all your scenrios with the same client builder.